config/config.go

```diff
@@ -1025,6 +1025,7 @@ func SetupViper(v *viper.Viper, filename string) {
 	v.SetDefault("adapters.synacormedia.endpoint", "http://{{.Host}}.technoratimedia.com/openrtb/bids/{{.Host}}")
 	v.SetDefault("adapters.tappx.endpoint", "http://{{.Host}}")
 	v.SetDefault("adapters.telaria.endpoint", "https://ads.tremorhub.com/ad/rtb/prebid")
+	v.SetDefault("adapters.trafficgate.endpoint", "http://{{.Host}}/?c=o&m=rtb")
```

Comment:
Hi. It need for my architect. I have 50 clusters with different domains. So, I have to create 50 same adapters for each where will change one domain only? What can you suggest me in my case? Can I heave it? It is the best solve for me.

Reply:
Can I leave it?*

Reply:
"config.Endpoint is the base url of your bidding server and may be interpreted as either a literal address or as a templated macro to support dynamic domains or dynamic paths."

This is from your documentation. I have dynamic domains. So, you support it.

Reply:
> What can you suggest me in my case?

Certainly. Can you please explain why you need to have 50 different clusters? Are these publisher specific? How does a publisher know which cluster to use? Are these entirely different domains or are they subdomains?

> Can I leave it?*

You may not. Let's continue discussing alternatives for your use case in this thread.

> This is from your documentation. 

Thank you for pointing this out. The comment is outdated and we'll push a fix. Please prefer our new adapter docs: https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html which includes the following:

"You may not use an endpoint domain as a bidder parameter. Prebid Server is not an open proxy. If absolutely necessary, you may specify a portion of the domain as a parameter to support geo regions or account specific servers. However, this is discouraged and may degrade the performance of your adapter since the server needs to maintain more outgoing connections. Host companies may choose to disable your adapter if it uses a dynamically configured domain."

Reply:
So, I have many partners that rent our product like White Label Solution. Each partner have own domain and cluster that allow them invite publishers and give them instructions for integration from Admin panel where they can get own domain and another parameters and even code example with the desired domain. 

Domain is not subdomain because our partners want be more separate from another partners and be uniq. It is very important like separate cluster!

So, we need dynamic domain (not subdomain) for this case. Because create 50 and more same adapters not rational and it will slow down the server and annoying them to approve

What are you think can we leave endpoint "http://{{.Host}}/?c=o&m=rtb" for this case? Because every our partners it like another DSP with own separate uniq demand.

Reply:
Hi. Any idea?

Reply:
It is from your documentation "If absolutely necessary, you may specify a portion of the domain as a parameter to support geo regions or account specific servers". I rewrite with subdomain "http://{{.Host}}.bc-plugin.com/?c=o&m=rtb" and rewrite tests for this. I hope you approve it.

Reply:
And I have 1 failing checks but https://github.com/prebid/prebid-server/commit/74d2199c7dd7e6b2c28d67c26cda2c4405ca5b6f it is not my commit.

Reply:
Hello. Why is my pull request have fail check "Code scanning results / Trivy"?

Reply:
Solved the problem with merging 'upstream/master'

Reply:
> So, I have many partners that rent our product like White Label Solution. Each partner have own domain and cluster that allow them invite publishers and give them instructions for integration from Admin panel where they can get own domain and another parameters and even code example with the desired domain.

Gotcha. Thank you for the context.

> It is from your documentation "If absolutely necessary, you may specify a portion of the domain as a parameter to support geo regions or account specific servers". I rewrite with subdomain "http://{{.Host}}.bc-plugin.com/?c=o&m=rtb" and rewrite tests for this. I hope you approve it.

Thus approach sounds good, provided that `bc-plugin.com` is a domain owned by your company redirecting for your clients and not a generic redirect service.

The other option is to use aliases, where each white label partner of yours would have their own configuration with whichever domain the setup in your system, along with their GVL ID and which media types they support - which can be a subset of this adapter. There's a little bit of code currently required and we have planned improvements to simplify in the future.

This alias approach is taken by other companies in a similar situation. The best example may be with the adtelligent aliases. Your customers may prefer this approach since they will receive top billing in the adapter list. The downsides are extra files required per adapter including a documentation entry. The subdomain approach you recommended would be simpler.

> Hello. Why is my pull request have fail check "Code scanning results / Trivy"?

Sorry about that. That's from a new security vulnerability. We've fixed it in the master branch, so merging in the latest was a good idea. If you see this again in the future and didn't change go.mod, no need to worry. This failure doesn't prevent merging.

Reply:
Hello. Yes, bc-plugin.com is a domain owned by your company. I want to stay on the option with subdomains. Thank you. When can you merge it?